ocaml/typing/uniqueness_analysis.ml

```diff
@@ -1020,26 +1040,39 @@ let rec pattern_match_tuple pat values =
 and pattern_match_single pat paths : Ienv.Extension.t * UF.t =
   let loc = pat.pat_loc in
   let occ = Occurrence.mk loc in
+  (* To read from the allocation, we need to borrow its memory cell
+     and set the unique_barrier. However, we do not read in every case,
+     since the user might want use a wildcard for already-consumed data. *)
+  let borrow_memory_address access =
+    Unique_barrier.enable pat.pat_unique_barrier;
+    match access with
+    | `NoRead -> UF.unused
```

Comment:
Is this the only use of `` `NoRead `` and `` `Read ``? If so, just make two functions: `no_borrow_memory_address` and `borrow_memory_address`, say.